src/views/List.tsx

```diff
@@ -1,21 +1,35 @@
-import React from 'react';
-import { ListItem as ListItemComponent } from '../components/ListItem';
-import { ListItem } from '../api';
+import { useState } from "react";
+import { ListItem as ListItemComponent } from "../components/ListItem";
+import { FilterList as FilterListComponent } from "../components/FilterList";
+import { ListItem } from "../api";
 
 interface Props {
 	data: ListItem[];
 }
 
 export function List({ data }: Props) {
```

Comment:
One thing I love to do in situations like this is rename props on the fly based on the context of how the component is using them. Sometimes its helpful to call the prop something simple like `data`, but inside the component that "data" has a different meaning: it's really `unfilteredListItems`. You can change the name of the variable when you deconstruct the `Props` object like this `List({ data: unfilteredListData })` which can bring new contextual meaning to your references of `data` without changing what the function signature looks like to the parent component: it just thinks of this as "data" and "you" get to think of it as what it rightfully is: unfiltered list data

Reply:
I love this and the ability to give the code more context!